### PR TITLE
Issue #860 identifier scoping documentation

### PR DESCRIPTION
**Pull request by Rene2mt**
# Committer Notes

Issue 860 updating OSCAL model descriptions where needed to clarify the scope and uniqueness of identifiers used within the OSCAL models.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/OSCAL/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.

### CODE REVIEW

**Review by brian-ruf — CHANGES_REQUESTED** [review]
I reviewed this at the request of @iMichaela, but paused my review based on a fundamental observation. I believe many of these descriptions do not **fully** describe the scope. 

For example, while it's true that each document-level UUID is ONLY document scoped, many other identifiers are both document scoped and intended to be visible by downstream documents as a result of `import` statements.

I suggest this be re-worked. Consider borrowing from programming parlance describing variable scope, such as "local" describing an identifier that is only intended to be used within the document itself and "global" describing an identifier that is intended to be used both within the document itself as well as by any document importing that document. 
There are only certain allowed import paths, so "global" would be constrained by import possibilities.  

NOTE: After five months I am a little rusty, but if there are any identifiers that are exposed to some documents through `import` statements, while they are not exposed to other documents despite being imported, we may need a more granular expression of scope beyond just "local" and "global". 

Ideally, I believe scope should reflect these concepts (as examples, not necessarily in this format):
**Identifier**|** Catalog**|**Profile**|**Component**|**SSP**|**AP**|**AR**|**POA&M**
:-----:|:-----:|:-----:|:-----:|:-----:|:-----:|:-----:|:-----:
Catalog UUID|X||||||
Catalog `control-id`|X|X|X|X|X|X|X
SSP Component UUID|||X|X|X|X|X
AP Subject UUID||||X|X|
AR Subject UUID|||||X|

Known import paths constraining "global" include:
- Catalog -> Profile -> Component Definition
- Catalog -> Profile -> SSP -> POA&M
- Catalog -> Profile -> SSP -> AP -> AR

Finally, while I haven't been in the working sessions in the past several months and don't know if there was discussion/consensus on terminology, I am not certain "document" is the best term to use to talk about scope. Consider using "model".

**Comment by Rene2mt** [comment]
Thanks for the inputs.  Please review the latest two commits to see the latest WIP ( https://github.com/usnistgov/OSCAL/pull/941/commits/81dfc3c5b7c93e58fc462397f019e8343c82fdc7 and https://github.com/usnistgov/OSCAL/commit/108b86ef7b893fd69f817dfefb607d175c8706a2 ).  These commits just have changes for identifier descriptions in just two of the metaschemas but additional changes will be made and committed once we confirm we are on the right track:
-  oscal_metadata_metaschema.xml
- oscal_catalog_metaschema.xml

As I reviewed the identifier descriptions in these metaschemas, I focused on clarifying

1. Uniqueness of the Identifier - I added "global" or "local" to describe the "uniqueness" of identifiers.  By default, identifiers of type uuid are globally unique, but it thinks its still good to call out explicitly.  Other identifiers (e.g. of type token) are often unclear, so I focused on these to make sure it was clearly stated.
2. Scope (of use) - I added some verbiage to indicate if the identifier was "cross-instance" scoped or just "instance" scoped.
3. Consistency - I made sure the descriptions specified what should happen to the identifier when revisions are made (e.g., should remain consistent across revisions or should change with each revision).

``

**Comment by david-waltermire-nist** [comment]
How about something like this:

```
<!-- identifier declarations -->
<description>A <em>machine-oriented</em>, <em>globally unique</em> identifier with a <em>cross-instance</em> scope that can be used to reference this defined resource elsewhere in this or another OSCAL instance. This UUID should be assigned <em>per-subject</em>, which means it should be consistently used to identify the same subject across revisions of the document.</description>
<remarks>
    <!-- UUID -->
    <p>Type: machine-oriented</p>
    <p>Uniqueness: globally-unique</p>
    <p>Scope: cross-instance - Provides an identifier that can be used to refer to this concept in OSCAL APs, ARs, and POAMs.</p>
    <p>Consistency: per-subject</p>
    <p>Additional Information:</p>
</remarks>
```
and for references:

```
<!-- identifier references -->
<description>Something similar to the above that incorporates key words</description>
<remarks>
    <!-- UUID -->
    <p>Type: machine-oriented</p>
    <p>Source: cross-instance - References an identifier from the SSP referenced in the assessment plan's <code>import-ssp</code> directive.</p>
    <p>Additional Information:</p>
</remarks>
```

File: docs/content/concepts/identifier-use/_index.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+---
+
+This page reviews important concepts to be aware of when declaring or referencing identifiers in OSCAL models, with in-depth explanations of identifier uniqueness and scope.  
+
+### **Identifier Type** 
+By design, OSCAL supports [*machine-oriented*](/concepts/identifier-use/#machine-oriented) and [*human-oriented*](/concepts/identifier-use/#human-oriented) identifiers. The OSCAL models dictate which are used for different data items.
+
+#### Machine-Oriented
+
+*Machine-oriented* identifiers are implemented as Universally Unique Identifier (UUID) as defined by [RFC 4122](https://tools.ietf.org/html/rfc4122) and are commonly used to (programmatically) identify or refer to unique information within OSCAL models.  Because OSCAL’s *`machine-oriented`* identifiers utilize UUID data types, they can be issued without a central authority, and used across OSCAL instances with low risk of identifier collisions.
```
Comment (david-waltermire-nist) [inline docs/content/concepts/identifier-use/_index.md:21]:
```suggestion
*Machine-oriented* identifiers provide a persistent identity for an entity within OSCAL the OSCAL models, which can be used in other locations within related OSCAL models to reference the associated entity. 

These identifiers are intended to be auto-generated by tools when the entity is initially created. In OSCAL, a machine-oriented identifier is implemented using a Universally Unique Identifier (UUID) as defined by [RFC 4122](https://tools.ietf.org/html/rfc4122). UUIDs were chosen because:

- Programming interfaces exist in most programming environments to generate a UUID
- UUIDs can be issued without a central authority
- UUIDs are represented in 128bits, providing for a large address space with low risk of identifier collisions for randomly generated values
```

File: docs/content/concepts/identifier-use/_index.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+This page reviews important concepts to be aware of when declaring or referencing identifiers in OSCAL models, with in-depth explanations of identifier uniqueness and scope.  
+
+### **Identifier Type** 
+By design, OSCAL supports [*machine-oriented*](/concepts/identifier-use/#machine-oriented) and [*human-oriented*](/concepts/identifier-use/#human-oriented) identifiers. The OSCAL models dictate which are used for different data items.
+
+#### Machine-Oriented
+
+*Machine-oriented* identifiers are implemented as Universally Unique Identifier (UUID) as defined by [RFC 4122](https://tools.ietf.org/html/rfc4122) and are commonly used to (programmatically) identify or refer to unique information within OSCAL models.  Because OSCAL’s *`machine-oriented`* identifiers utilize UUID data types, they can be issued without a central authority, and used across OSCAL instances with low risk of identifier collisions. 
+
+> **NOTE**: While UUIDs are used as machine-oriented identifiers catering to use cases where automated processing is desired, they are not ideal for wildcard matching scenarios.  UUIDs should only be used when exact matches are expected.
```
Comment (david-waltermire-nist) [inline docs/content/concepts/identifier-use/_index.md:23]:
```suggestion
The opaque nature of UUIDs, which consist of a series of hexadecimal characters, makes them less than ideal for wildcard matching scenarios.  Thus, their use in OSCAL is intended for identification only where an exact match is required. Where wildcard matching is needed, the other data elements associated with the entity should be evaluated for a match instead. 
```

File: docs/content/concepts/identifier-use/_index.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+
+The [OSCAL Reference Index](/reference/latest/complete/xml-index/#/@uuid) provides a complete listing of UUIDs in OSCAL.  References to these identifiers typically follow a naming convention of the object type followed by “-uuid”.  For example, see the reference index for [location-uuid](/reference/latest/complete/xml-index/#/location-uuid).
```
Comment (david-waltermire-nist) [inline docs/content/concepts/identifier-use/_index.md:26]: This is a link to the XML index. The JSON index should be referenced as well.

File: docs/content/concepts/identifier-use/_index.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+#### Human-Oriented
+
+OSCAL generally implements *`human-oriented`* identifiers as Token data types, which are non-colonized names (add link to data type).  These *`human-oriented`* identifiers have the benefit of allowing authors to incorporate semantics that support readability and processing by humans.  For example, control identifiers in a catalog may use a nomenclature that is familiar to the intended audience, allowing them to quickly determine what security control is being referred to, simply by its ID.
```
Comment (david-waltermire-nist) [inline docs/content/concepts/identifier-use/_index.md:30]:
```suggestion
OSCAL generally implements *`human-oriented`* identifiers as Token data types, which are non-colonized names (add link to data type).  These *`human-oriented`* identifiers have the benefit of allowing authors to incorporate semantics that support readability and processing by humans.  For example, control identifiers in a catalog may use a nomenclature that is familiar to the intended audience, allowing them to quickly determine what security control is being referred to, simply by its identifier value.  
```

File: docs/content/concepts/identifier-use/_index.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+#### Human-Oriented
+
+OSCAL generally implements *`human-oriented`* identifiers as Token data types, which are non-colonized names (add link to data type).  These *`human-oriented`* identifiers have the benefit of allowing authors to incorporate semantics that support readability and processing by humans.  For example, control identifiers in a catalog may use a nomenclature that is familiar to the intended audience, allowing them to quickly determine what security control is being referred to, simply by its ID.  
+
+The [OSCAL Reference Index](/reference/latest/complete/xml-index/#/@id) provides a comprehensive listing of the *`human-oriented`* IDs in OSCAL.  References to these IDs are typically named according to the referenced object type (e.g., control) followed by “-id”, as seen here in the [Reference Index](/reference/latest/complete/xml-index/#/@control-id).
```
Comment (david-waltermire-nist) [inline docs/content/concepts/identifier-use/_index.md:32]: Need both XML and JSON references here.

File: docs/content/concepts/identifier-use/_index.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+### **Uniqueness** 
+OSCAL identifier uniqueness are categorized as *`locally-unique`* or *`globally-unique`*.  As implied by the category name, *`locally-unique`* identifiers must be unique within the current document whereas *`globally-unique`* identifiers are guaranteed to be unique across all other identifiers.  OSCAL’s *`machine-oriented`* UUID identifiers are always *`globally-unique`*.  *`Human-oriented`* identifiers must be defined and managed organizationally and are more susceptible to identifier duplication or collisions. Thus, *`human-oriented`* identifiers are less likely to be *`globally-unique`*.
```
Comment (david-waltermire-nist) [inline docs/content/concepts/identifier-use/_index.md:35]:
```suggestion
OSCAL identifier uniqueness is categorized as *`locally-unique`* or *`globally-unique`*.  As implied by the category name, *`locally-unique`* identifiers must be unique within the current document, whereas *`globally-unique`* identifiers are guaranteed to be unique across all other identifiers.  OSCAL’s *`machine-oriented`* UUID identifiers are always *`globally-unique`*.  *`Human-oriented`* identifiers must be defined and managed organizationally and are more susceptible to identifier duplication or collisions. Thus, *`human-oriented`* identifiers are less likely or cannot be gauranteed to be *`globally-unique`*.  
```

**Review by aj-stein-nist — COMMENTED** [review]
Some initial feedback, more to come. I see Dave is adding a lot too, so I wanted to get this in the mix. Will give it more reviews a few times over before I say I am done done. :-)

File: docs/content/concepts/identifier-use/_index.md (hunk 3 above)
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md:34]: added JSON index.  Will be in next commit.

File: docs/content/concepts/identifier-use/_index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md]: Added link. will be in next commit

File: docs/content/reference/latest/assessment-plan/_index.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Assessment Plan Model v1.0.0 Reference"
+title: "Assessment Plan Model v Reference"
```
Comment (bradh) [inline docs/content/reference/latest/assessment-plan/_index.md:2]: Why this change?

File: docs/content/concepts/identifier-use/_index.md (hunk 5 above)
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md:40]: will update in next commit

File: docs/content/concepts/identifier-use/_index.md [hunk 8]
```diff
@@ -18,7 +18,15 @@ By design, OSCAL supports [*machine-oriented*](/concepts/identifier-use/#machine
 
 #### Machine-Oriented
 
-*Machine-oriented* identifiers are implemented as Universally Unique Identifier (UUID) as defined by [RFC 4122](https://tools.ietf.org/html/rfc4122) and are commonly used to (programmatically) identify or refer to unique information within OSCAL models.  Because OSCAL’s *`machine-oriented`* identifiers utilize UUID data types, they can be issued without a central authority, and used across OSCAL instances with low risk of identifier collisions. 
+*Machine-oriented* identifiers provide a persistent identity for an entity within the OSCAL models, which can be used in other locations within related OSCAL models to reference the associated entity. 
+
+These identifiers are intended to be auto-generated by tools when the entity is initially created. In OSCAL, a machine-oriented identifier is implemented using a Universally Unique Identifier (UUID) as defined by [RFC 4122](https://tools.ietf.org/html/rfc4122). A UUID is represented in OSCAL using the UUID datatype (provide link).
```
[the remark this answers is not in the document]
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md:23]: Agreed.  I'll add link to UUID data type with next commit

File: docs/content/concepts/identifier-use/_index.md (hunk 3 above)
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md:34]: valid point.  Updating locally and will include with my next commit

File: docs/content/concepts/_index.md [hunk 9]
```diff
@@ -18,6 +18,7 @@ This section of the OSCAL website presents:
 
 - Key [terminology](terminology/) used in OSCAL;
 - An overview of the OSCAL [layers and models](layer/), to include who and what processes they apply to;
+- An oververview of [identifier use](identifier-use) in OSCAL models;
```
[the remark this answers is not in the document]
Comment (Rene2mt) [reply docs/content/concepts/_index.md:21]: this is resolved now

File: docs/content/concepts/identifier-use/_index.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+UUIDs were chosen because:
+
+- Programming interfaces exist in most programming environments to generate a UUID
+- UUIDs can be issued without a central authority
+- UUIDs are represented in 128bits, providing for a large address space with low risk of identifier collisions for randomly generated values
+
+The opaque nature of UUIDs, which consist of a series of hexadecimal characters, makes them less than ideal for wildcard matching scenarios.  Thus, their use in OSCAL is intended for identification only where an exact match is required. Where wildcard matching is needed, the other data elements associated with the entity should be evaluated for a match instead. 
+
```
[the remark this answers is not in the document]
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md:32]: Removed the extra line locally.  willl include in next commit

File: docs/content/concepts/identifier-use/_index.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+### **Scope** 
+Identifiers are declared so that data items in OSCAL instances can be referenced accurately and consistently.  In the most basic instance, references are made to data items in the same OSCAL instance, but because OSCAL supports instance importing, there are scenarios where data items in an external / imported OSCAL instance need to be referenced as well.  Identifiers that are only intended for use within the same OSCAL instance are categorized as *`instance`* scope, but identifiers that can be referenced (from other OSCAL instances) are considered *`cross-instance`* scoped. While *`human-oriented`* identifiers might not be globally unique, they may be *`cross-instance`* scoped.  In these cases, the appropriate import attribute (e.g. import, import-component-definitions, import-profile, import-ssp, and import-ap) should be used (similar to a namespacing) to deconflict identifiers with the same values in the associated OSCAL instances.  This would allow for the same control IDs to be used and referenced in a profile and its imported catalog(s) for example.  The tables below provides a summary of OSCAL unique identifier scoping.
```
[the remark this answers is not in the document]
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md:46]: I reworked the scope section quite a bit locally, adding an svg diagram illustrating the relationship between OSCAL instances.  Links to import types would be useful...I'll go ahead and add them as well.

File: docs/content/concepts/identifier-use/_index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: I agree per you review of my review. I meant more in the aspirational view that they should not be reused intentionally by developers unless in clear case; UUIDs are presumed to not be reusable. In a previous part of my review, I thought somewhere here, I added [a link to the explanation of how UUIDs should be used for this reason](https://pages.nist.gov/OSCAL/concepts/layer/overview/#use-of-version-4-uuids).

I guess your point stands, maybe it is better not to muddy the waters! I'll resolve this one for now.

File: docs/content/concepts/identifier-use/_index.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+
+The opaque nature of UUIDs, which consist of a series of hexadecimal characters, makes them less than ideal for wildcard matching scenarios.  Thus, their use in OSCAL is intended for identification only where an exact match is required. Where wildcard matching is needed, the other data elements associated with the entity should be evaluated for a match instead. 
+
+
+The [OSCAL Reference Index](/reference/latest/complete/xml-index/#/@uuid) provides a complete listing of UUIDs in OSCAL.  References to these identifiers typically follow a naming convention of the object type followed by “-uuid”.  For example, see the reference index for [location-uuid](/reference/latest/complete/xml-index/#/location-uuid).
+
+#### Human-Oriented
+
+A *`human-oriented`* identifier incorporates semantics that support readability and processing by humans. OSCAL implements *`human-oriented`* identifiers as Token data types, which are non-colonized names (add link to data type). For example, control identifiers in a catalog may use a nomenclature that is familiar to the intended audience, allowing them to quickly determine what security control is being referred to, simply by its identifier value.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:38]: In some places we say it is a "token" data type and other places we capitalize and say a "Token" data type. We should be consistent, I am not sure what the team has agreed upon.

File: docs/content/concepts/identifier-use/_index.md (hunk 3 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md:34]: This will get crowded with brad's recommendation. It seems we single space between sentences and there is an additional space at "followed by “-uuid”.  For example," between the period and for. I noticed this when re-reading locally.

File: docs/content/concepts/identifier-use/_index.md [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+
+#### **Catalog Identifiers**
+Identifiers defined in a catalog may be referenced locally or from an importing profile.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:49]:
```suggestion
Identifiers defined in a catalog may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+|Catalog|Property uuid|&#9745;|&#9745;||||||
+|Catalog|Document Identifier scheme|&#9745;|&#9745;||||||
+|Catalog|Part id|&#9745;|&#9745;||||||
+|Catalog|Param id|&#9745;|&#9745;||||||
+|Catalog|Catalog uuid|&#9745;|&#9745;||||||
+|Catalog|Control id|&#9745;|&#9745;||||||
+|Catalog|Group id|&#9745;|&#9745;||||||
+
+#### **Profile Identifiers**
+Identifiers defined in a profile may be referenced locally or from an importing profile or SSP.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:66]:
```suggestion
Identifiers defined in a profile may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+|Profile|Resource uuid||&#9745;||&#9745;||||
+|Profile|Property uuid||&#9745;||&#9745;||||
+|Profile|Document Identifier scheme||&#9745;||&#9745;||||
+|Profile|Part id||&#9745;||&#9745;||||
+|Profile|Param id||&#9745;||&#9745;||||
+|Profile|Profile uuid||&#9745;||&#9745;||||
+|Profile|Group id||&#9745;||&#9745;||||
+
+#### **Component Definition Identifiers**
+Identifiers defined in a component definition may be referenced locally or from an importing component definition instance.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:82]:
```suggestion
Identifiers defined in a component definition may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 16]
```diff
@@ -0,0 +1,212 @@
...
+|Component Definition|Part id|||&#9745;|||||
+|Component Definition|Param id|||&#9745;|||||
+|Component Definition|Component Definition uuid|||&#9745;|||||
+|Component Definition|Component uuid|||&#9745;|||||
+|Component Definition|Capability uuid|||&#9745;|||||
+|Component Definition|Control Implementation Set uuid|||&#9745;|||||
+|Component Definition|Control Implementation uuid|||&#9745;|||||
+
+#### **SSP Identifiers**
+Identifiers defined in an SSP may be referenced locally or from an importing AP or POA&M.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:101]:
```suggestion
Identifiers defined in an SSP may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+|SSP|Leveraged Authorization uuid||||&#9745;|&#9745;||&#9745;|
+|SSP|Control Requirement uuid||||&#9745;|&#9745;||&#9745;|
+|SSP|By-Component uuid||||&#9745;|&#9745;||&#9745;|
+|SSP|Provided uuid||||&#9745;|&#9745;||&#9745;|
+|SSP|Responsibility uuid||||&#9745;|&#9745;||&#9745;|
+|SSP|Inherited uuid||||&#9745;|&#9745;||&#9745;|
+|SSP|Satisfied uuid||||&#9745;|&#9745;||&#9745;|
+
+#### **AP Identifiers**
+Identifiers defined in an AP may be referenced locally or from an importing AR.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:126]:
```suggestion
Identifiers defined in an AP may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+|AP|Mitigating Factor uuid|||||&#9745;|&#9745;||
+|AP|Implementation uuid|||||&#9745;|&#9745;||
+|AP|Risk Log Entry uuid|||||&#9745;|&#9745;||
+|AP|Remediation uuid|||||&#9745;|&#9745;||
+|AP|Required Asset uuid|||||&#9745;|&#9745;||
+|AP|Part uuid|||||&#9745;|&#9745;||
+|AP|Assessment Plan uuid|||||&#9745;|&#9745;||
+
+#### **AR Identifiers**
+Identifiers defined in an AR may be referenced locally or from an importing POA&M.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:153]:
```suggestion
Identifiers defined in an AR may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+|AR|Remediation uuid||||||&#9745;|&#9745;|
+|AR|Required Asset uuid||||||&#9745;|&#9745;|
+|AR|Part uuid||||||&#9745;|&#9745;|
+|AR|Assessment Results uuid||||||&#9745;|&#9745;|
+|AR|Assessment Result uuid||||||&#9745;|&#9745;|
+|AR|Assessment Log Entry uuid||||||&#9745;|&#9745;|
+|AR|Finding uuid||||||&#9745;|&#9745;|
+
+#### **POA&M Identifiers**
+Identifiers defined in a POA&M may be referenced locally.
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:183]:
```suggestion
Identifiers defined in a POA&M may be referenced locally or other OSCAL document instances of different models. For a given identifier in a row, columns with a &#9745; indicate possible locations for cross-referencing.
```

File: docs/content/concepts/identifier-use/_index.md [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+
+#### **Catalog Identifiers**
+Identifiers defined in a catalog may be referenced locally or from an importing profile.
+
+|**Defining Model**|**Identifier**|**Catalog**|**Profile**|**Component**|**SSP**|**AP**|**AR**|**POA&M**|
```
Comment (aj-stein-nist) [inline docs/content/concepts/identifier-use/_index.md:51]: General feedback on the headers, we might want to consider saying, like in this catalog example:

```suggestion
|**Defining Model**|**Identifier**|**Self**|**Profile**|**Component**|**SSP**|**AP**|**AR**|**POA&M**|
```

or

```suggestion
|**Defining Model**|**Identifier**|**Catalog (Self)**|**Profile**|**Component**|**SSP**|**AP**|**AR**|**POA&M**|
```

Either that, or we mean "this document instance, but other catalogs can cross-reference each other!" but I am not sure 1) what is possible for each of the document instances and 2) if we want to muddy the waters and bring that up. Does that make sense?

**Review by aj-stein-nist — COMMENTED** [review]
Some more feedback, hope this helps. I am free to talk today if you want to work on it together.

File: docs/content/concepts/identifier-use/_index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: OK, I moved the larger discussion into usnistgov/OSCAL#1105.

As for the content here, my recommendation below: OSCAL generally provides the token data to enable *`human-oriented`* identifiers.  These *`human-oriented`* identifiers ..."

File: docs/content/concepts/identifier-use/_index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (Rene2mt) [reply docs/content/concepts/identifier-use/_index.md]: we'll create a follow on issue for this item

File: docs/content/concepts/identifier-use/_index.md (hunk 12 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: Dave said we should keep them lower case. This still might need to be fixed, @Rene2mt.

File: docs/content/concepts/identifier-use/_index.md (hunk 15 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 13 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 14 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 16 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 17 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 18 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 20 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/concepts/identifier-use/_index.md (hunk 19 above)
Comment (aj-stein-nist) [reply docs/content/concepts/identifier-use/_index.md]: This is obsolete now.

File: docs/content/reference/latest/assessment-plan/_index.md (hunk 7 above)
Comment (aj-stein-nist) [reply docs/content/reference/latest/assessment-plan/_index.md]: This is based in some automation issues with the model documentation script. We discussed this during a PR review meeting with Dave and Rene. I will open a bug for this to make sure it gets tracked later.

Comment (aj-stein-nist) [reply docs/content/reference/latest/assessment-plan/_index.md]: Ok opened up usnistgov/OSCAL#1106 to track this specifically and move a fix outside of this PR conversation. Hope that clarifies, Brad.
